test/helper.go

```diff
@@ -0,0 +1,277 @@
+package test
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	"github.com/Kuadrant/multicluster-gateway-controller/pkg/_internal/conditions"
+	"github.com/Kuadrant/multicluster-gateway-controller/pkg/apis/v1alpha1"
+	"github.com/Kuadrant/multicluster-gateway-controller/pkg/placement"
+)
+
+const (
+	TestDomain             = "thecat.com"
+	ValidTestHostname      = "boop." + TestDomain
+	FailFetchDNSsubdinain  = "failfetch"
+	FailCreateDNSSubdomain = "failcreate"
+	FailEnsureCertHost     = "failCreateCert" + "." + TestDomain
+	FailGetCertSecretHost  = "failGetCert" + "." + TestDomain
+	FailEndpointsHostname  = "failEndpoints" + "." + TestDomain
+	FailPlacementHostname  = "failPlacement" + "." + TestDomain
+	TestCluster            = "test_cluster_one"
+	TestNamespace          = "boop-namespace"
+	TestDummyCRName        = "boop"
+	TestPlacement          = "placement"
+)
+
+func VerifyTLSSecretTestResultsAsExpected(got []v1.Object, want []v1.Object, gateway *v1beta1.Gateway) bool {
+	for _, wantSecret := range want {
+		match := false
+		for _, gotSecret := range got {
+			if wantSecret.GetName() == gotSecret.GetName() &&
+				reflect.DeepEqual(wantSecret.GetLabels(), gotSecret.GetLabels()) &&
+				reflect.DeepEqual(wantSecret.GetAnnotations(), gotSecret.GetAnnotations()) &&
+				wantSecret.GetNamespace() == gateway.GetNamespace() {
+				match = true
+				break
+			}
+		}
+		if !match {
+			return false
+		}
+	}
+	return true
+}
+
+func GetValidTestRequest() ctrl.Request {
+	return ctrl.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      TestDummyCRName,
+			Namespace: TestNamespace,
+		},
+	}
+}
```

Comment:
Instead of Get, Build might be a better name for these since you are building a new instance of a resource. Get suggests you are getting it via the API.

I'd also suggest if you want these to be reusable outside of your tests that you add params to a function to let you change appropriate resource values. 

```
func BuildValidTestRequest() ctrl.Request {
	return BuildTestRequest(TestDummyCRName, TestNamespace)
}

func BuildTestRequest(name, ns string) ctrl.Request {
	return ctrl.Request{
		NamespacedName: types.NamespacedName{
			Name:      name,
			Namespace: ns,
		},
	}
}
```

Reply:
renamed to `build...`